### PR TITLE
try hyperx

### PR DESCRIPTION
**Pull request by ngoldman**
This is a test branch to show what working with https://github.com/substack/hyperx would like.

Also under discussion is using https://github.com/Raynos/main-loop to optimize virtual dom updates.

### CODE REVIEW

**Comment by ngoldman** [comment]
rebasing...

File: renderer/views/torrent-list.js [hunk 1]
```diff
@@ -19,51 +21,75 @@ function renderTorrent (state, dispatch, torrent) {
...
-      className: !torrent.ready ? 'disabled' : '',
-      onclick: () => dispatch('openAirplay', torrent)
-    }, 'airplay'))
+    elements.push(hx`
+      <i
+        className="${!torrent.ready ? 'disabled btn.icon.airplay' : 'btn.icon.airplay'}"
+        onclick=${() => dispatch('openAirplay', torrent)}>
+        airplay
+      </i>
+    `)
```
Comment (ngoldman) [inline renderer/views/torrent-list.js]: Not super happy with how this looks compared to plain `h` but it seemed liked the best way to translate the items in the `elements` array into hyperx.

Comment (ngoldman) [inline renderer/views/torrent-list.js]: actually this is wrong, one more rebase on its way

Comment (feross) [inline renderer/views/torrent-list.js]: is there no way to include some classes in the element name, like a css selector?

`i.btn.icon.airplay`

**Comment by ngoldman** [comment]
Okay pulled in changes from your last two PRs, tested locally, everything looks good to me. Player changes look nice :+1:

File: renderer/views/torrent-list.js (hunk 1 above)
Comment (ngoldman) [inline renderer/views/torrent-list.js]: @feross It's imitating html like so:

```
<element attr="" attr2="${}" fn=${}></element>
```

As far as I know that's as far as it goes.

Comment (ngoldman) [inline renderer/views/torrent-list.js]: Probably the smart thing to do is just pull the ternary out into a function.

Comment (feross) [inline renderer/views/torrent-list.js]: Actually, looks like this would work!

```js
hx`<div.hi>`
// [ 'div.hi', {}, [] ]
```

Comment (ngoldman) [inline renderer/views/torrent-list.js]: whoa, weird!

File: renderer/views/torrent-list.js [hunk 2]
```diff
@@ -19,51 +21,74 @@ function renderTorrent (state, dispatch, torrent) {
...
   ]
+
   if (state.view.chromecast) {
-    elements.push(h('i.btn.icon.chromecast', {
-      className: !torrent.ready ? 'disabled' : '',
-      onclick: () => dispatch('openChromecast', torrent)
-    }, 'cast'))
+    elements.push(hx`
+      <i.btn.icon.chromecast
+        className="${!torrent.ready ? 'disabled' : ''}"
```
Comment (ngoldman) [inline renderer/views/torrent-list.js]: this notation works and is a little less of an eyesore

Comment (feross) [inline renderer/views/torrent-list.js]: nice!

**Comment by feross** [comment]
LGTM. Let's merge and use it for a while and see how we like it :)

**Pull request merged**